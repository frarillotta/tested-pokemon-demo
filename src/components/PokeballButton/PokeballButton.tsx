import { Pokeball } from '../SVGs';
import styled from "styled-components";
import { CSSProperties, forwardRef } from 'react';

type PokeballButtonProps = {
    isCaptured: boolean;
    onClick: () => void;
    size?: number;
    style?: CSSProperties
}

const PokeballButton = forwardRef(({isCaptured, onClick, size = 24, ...delegate}: PokeballButtonProps, ref) => {
	return <PokeballWrapper 
		onClick={onClick}
		ref={ref}
		data-t-is-captured={isCaptured}
		{...delegate}
	>
		<PokeballSVG 
			size={size}
			isCaptured={isCaptured}
		/>
	</PokeballWrapper>
})
PokeballButton.displayName = "PokeballButton";

const PokeballWrapper = styled.div`
	cursor: pointer;
`

const PokeballSVG = styled(Pokeball)`
	filter: ${({isCaptured}) => isCaptured ? "" : "opacity(0.3)"};
`

export { PokeballButton }