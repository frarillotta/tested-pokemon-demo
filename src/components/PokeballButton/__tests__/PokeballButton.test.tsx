import { render } from "@testing-library/react";
import {PokeballButton} from "../PokeballButton";

describe("PokeballButton", () => {
	
	describe("when isCaptured=true", () => {
		
		it("should have a consistent snapshot", () => {
			const {container} = render(<PokeballButton isCaptured={true} onClick={jest.fn()} />);
			expect(container.querySelector("[data-t-is-captured]").getAttribute("data-t-is-captured")).toBe("true");
			expect(container).toMatchSnapshot();
		});

	});

	describe("when isCaptured=false", () => {
		
		it("should have a consistent snapshot", () => {
			const {container} = render(<PokeballButton isCaptured={false} onClick={jest.fn()}/>);
		
			expect(container.querySelector("[data-t-is-captured]").getAttribute("data-t-is-captured")).toBe("false");
			expect(container).toMatchSnapshot();
		})
	})
	
})