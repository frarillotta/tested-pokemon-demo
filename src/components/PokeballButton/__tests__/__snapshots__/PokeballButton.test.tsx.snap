// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeballButton when isCaptured=false should have a consistent snapshot 1`] = `
<div>
  <div
    class="PokeballButton__PokeballWrapper-sc-z0qx2i-0 jUctph"
    data-t-is-captured="false"
  >
    <svg
      class="PokeballButton__PokeballSVG-sc-z0qx2i-1 fWWEza"
      height="24"
      viewBox="25 25 50 50"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M 30 50
 a 1 1 1 0 1 40 0
 h-12.5
 a 1 1 1 0 0 -15 0
 z"
        fill="#f00"
        stroke="#222"
      />
      <circle
        cx="50"
        cy="50"
        fill="#222"
        r="5"
        stroke="#222"
      />
      <path
        d="M 30 50
 a 1 1 1 0 0 40 0
 h-12.5
 a 1 1 1 0 1 -15 0
 z"
        fill="#fff"
        stroke="#222"
      />
    </svg>
  </div>
</div>
`;

exports[`PokeballButton when isCaptured=true should have a consistent snapshot 1`] = `
<div>
  <div
    class="PokeballButton__PokeballWrapper-sc-z0qx2i-0 jUctph"
    data-t-is-captured="true"
  >
    <svg
      class="PokeballButton__PokeballSVG-sc-z0qx2i-1 iJvgOH"
      height="24"
      viewBox="25 25 50 50"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M 30 50
 a 1 1 1 0 1 40 0
 h-12.5
 a 1 1 1 0 0 -15 0
 z"
        fill="#f00"
        stroke="#222"
      />
      <circle
        cx="50"
        cy="50"
        fill="#222"
        r="5"
        stroke="#222"
      />
      <path
        d="M 30 50
 a 1 1 1 0 0 40 0
 h-12.5
 a 1 1 1 0 1 -15 0
 z"
        fill="#fff"
        stroke="#222"
      />
    </svg>
  </div>
</div>
`;
